test/jdk/com/sun/jdi/ThreadMemoryLeakTest.java

```diff
@@ -27,7 +27,7 @@
  * @summary JDI memory leak when creating and destroying many threads
  *
  * @comment Don't allow -Xcomp or -Xint as they impact memory useage and number of iterations
- * @requires (vm.compMode == "Xmixed")
+ * @requires (vm.compMode == "Xmixed") & !(vm.gc.Z & vm.opt.final.ZGenerational)
```

Comment:
When I first wrote this test, it ended up failing with ZGC because I hadn't tested it. I considered excluding it for the same reason you've given, but then considered that the test might expose a leak with one GC, but not others, so I decided to fix it.

Reply:
I've created JDK-8307402. I'll push a problem list entry.